controller/GiPlanned.controller.js

```diff
@@ -0,0 +1,408 @@
+		 * @memberOf smud.org.wmLM01.view.GiPlanned
+		 */
+		//onExit: function() {
+
+		//},
+		getItems: function() {
+			var oTable = sap.ui.getCore().byId(zController.getView().createId("materialTable"));
+			var oModel = zController.getView().getModel();
+			var data = zController.getView().getModel().getData();
+			oTable.removeAllItems();
+			if (oTable.getBinding("items")) {
+				oTable.getBinding("items").refresh();
+			}
+			oModel.setProperty("/Benum", "");
+			if (!data.Tanum) {
+				MessageBox.show("Enter a Transfer Order", {
+					title: "Error",
+					icon: "ERROR"
+				});
+				return;
+			}
+			// Busy dialog...
+			var busy = new sap.m.BusyDialog({
+				text: "Retrieving transfer order items to confirm"
+			});
+			busy.open();
+
+			// Filters for Entity Set
+			var filters = [
+				new sap.ui.model.Filter({
+					path: "Lgnum",
+					operator: sap.ui.model.FilterOperator.EQ,
+					value1: data.Lgnum
+				}),
+				new sap.ui.model.Filter({
+					path: "Tanum",
+					operator: sap.ui.model.FilterOperator.EQ,
+					value1: data.Tanum
+				})
+			];
+
+			var oDataModel = this.getView().getModel("GiPlannedService");
+			oDataModel.read("/ItemsSet", {
+				filters: filters,
+				success: function(oResult) {
+					var firstRow = oResult.results[0];
+					var oData = oResult.results;
+					oModel.setProperty("/Benum", firstRow.Benum);
+
+					for (var i = 0; i < oData.length; i++) {
+						oData[i].VsolmConfirm = "";
+					}
+
+					// Bind the results to the material table
+					var colItemTemplate = sap.ui.getCore().byId(zController.getView().createId("tableColumnListItem"));
+					var tabModel = new sap.ui.model.json.JSONModel(oData, false);
+					tabModel.setData(oData, false);
+					var context = new sap.ui.model.Context(tabModel, "/");
+					oTable.setBindingContext(context);
+					oTable.setModel(tabModel);
+					oTable.bindItems("/", colItemTemplate);
+
+					busy.close();
+				},
+				error: function(oError) {
+					MessageBox.show(JSON.parse(oError.responseText).error.message.value, {
+						title: "Error",
+						icon: "ERROR"
+					});
+					busy.close();
+				}
+			});
+		},
+
+		scan: function(oEvent) {
+			GlobalFuncs.scan(oEvent);
+		},
+		scanTanum: function(oEvent) {
+			GlobalFuncs.scan(oEvent, zController.getItemsAfterScan);
+		},
+		getItemsAfterScan: function() {
+			if (zController.getView().getModel().getProperty("/Tanum")) {
+				zController.getItems();
+			}
+		},
+		createTable: function() {
+			// Create the item table via JS
+			var oTable = new sap.m.Table(zController.getView().createId("materialTable"), {
+				headerText: "Confirm Items",
+				width: "100%",
+				growing: true,
+				filter: zController.handleMaterialSearch,
+				alternateRowColors: true,
+				growingThreshold: 50,
+				noDataText: "No items selected to confirm",
+				type: "Active" //enables itemPress event
+			});
+
+			var toolbar = new sap.m.Toolbar({

```

Comment:
same here, why are we creating all the control in JS, can't we define them in the XML view?